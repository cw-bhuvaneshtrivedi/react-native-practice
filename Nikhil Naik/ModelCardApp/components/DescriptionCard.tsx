import { View, Text, StyleSheet, Image } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
import { DescriptionProp } from "./Types/Type";

export const DescriptionCard = ({ img, maker, model }: DescriptionProp) => {
  return (
    <View style={style.container}>
      <Image
        source={{
          uri: img,
        }}
        style={style.image}
        testID="carImage"
      />
      <View style={style.name}>
        <Text style={style.maker} testID="makerName">
          {maker}
        </Text>
        <Text style={style.model} testID="modelName">
          {model}
        </Text>
      </View>
      <View style={style.arrow}>
        <MaterialIcons name="arrow-forward-ios" size={24} color="gray" />
      </View>
    </View>
  );
};

const style = StyleSheet.create({
  container: {
    width: 301,
    paddingBottom: 21,
    paddingTop: 20,
    paddingLeft: 15,
    flexDirection: "row",
    justifyContent: "flex-start",
  },
  image: {
    width: 130,
    height: 73,
  },
  maker: {
    fontSize: 13,
    color: "#aaaaaa",
  },
  model: {
    fontSize: 18,
    color: "#484848",
    paddingTop: 4,
  },
  name: {
    paddingTop: 16,
    paddingBottom: 15,
    paddingLeft: 16,
  },
  arrow: {
    marginRight: 0,
    marginLeft: "auto",
    alignSelf: "center",
  },
});
