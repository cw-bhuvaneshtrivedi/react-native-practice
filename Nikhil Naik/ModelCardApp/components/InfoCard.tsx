import {
  StyleSheet,
  View,
  Text,
  Image,
  Pressable,
  TouchableOpacity,
} from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { FontAwesome } from "@expo/vector-icons";
import { useState } from "react";

export const InfoCard = ({ ver, border }: any) => {
  const [save, setSave] = useState(false);
  const [compare, setCompare] = useState(false);
  return (
    <View style={style.container}>
      <View style={style.containerMain}>
        <Text style={style.containerName} testID="carName">
          {ver.name}
        </Text>
        <View style={style.carDescriptionContainer}>
          <Text style={style.carDescriptionContainerText}>{ver.maxPower}</Text>
          <Text style={style.carDescriptionContainerText}>
            {ver.displacement}
          </Text>
          <Text style={style.carDescriptionContainerText}>
            {ver.transmission}
          </Text>
          <Text style={style.carDescriptionContainerText}>{ver.fuelType}</Text>
        </View>
        <View>
          <View style={style.carPriceContainerMain}>
            <Text style={style.carPriceMainText}>
              {ver.priceOverview.price}
            </Text>
            <Text style={style.carPriceSecondaryText}>
              Show price in my city
            </Text>
            <Text style={style.carPriceEMIText}>EMI</Text>
          </View>
          <View style={style.carPriceContainerSecondary}>
            <Text style={style.carShowRoomPriceText}>
              Avg.Ex-Showroom Price
            </Text>
            <Text style={{ fontStyle: "italic" }}>{ver.emi}</Text>
          </View>
        </View>
      </View>
      <View style={style.containerOptions}>
        <TouchableOpacity
          onPress={(e) => {
            setSave(!save);
            border((count: number) => {
              return !save ? count + 1 : count - 1;
            });
          }}
          style={{ justifyContent: "center" }}
          testID="shortlistOptionButton"
        >
          <View
            style={style.containerOptionsElementLeft}
            testID="shortlistOption"
          >
            {save ? (
              <FontAwesome name="bookmark" size={15} color="#00afa0" />
            ) : (
              <FontAwesome name="bookmark-o" size={15} color="#6f6f6f" />
            )}
            {save ? (
              <Text style={style.containerOptionsText}>Shortlisted</Text>
            ) : (
              <Text style={style.containerOptionsText}>Shortlist</Text>
            )}
          </View>
        </TouchableOpacity>
        <View style={style.containerOptionsElementCenter}></View>
        <TouchableOpacity
          onPress={(e) => setCompare(!compare)}
          style={{ justifyContent: "center" }}
          testID="compareOptionButton"
        >
          <View
            style={style.containerOptionsElementRight}
            testID="compareOption"
          >
            <FontAwesome
              name="balance-scale"
              size={13}
              color={compare ? "#00afa0" : "#6f6f6f"}
            />
            {compare ? (
              <Text style={style.containerOptionsText}>
                Remove from Compare
              </Text>
            ) : (
              <Text style={style.containerOptionsText}>Add to Compare</Text>
            )}
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const style = StyleSheet.create({
  containerMain: {
    paddingLeft: 12,
    paddingRight: 12,
  },
  carShowRoomPriceText: {
    fontSize: 12,
    color: "#aaaaaa",
    alignSelf: "flex-end",
  },
  carEMIPrice: {
    color: "#484848",
    fontSize: 13,
  },
  carPriceMainText: {
    fontSize: 16,
    color: "#484848",
    fontWeight: "bold",
  },
  carPriceSecondaryText: {
    fontSize: 13,
    color: "#0288d1",
    alignSelf: "flex-end",
  },
  carPriceEMIText: {
    fontSize: 13,
    color: "#aaaaaa",
    alignSelf: "flex-end",
    fontStyle: "italic",
  },
  carPriceContainerMain: {
    paddingTop: 16,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  carPriceContainerSecondary: {
    paddingTop: 4,
    flexDirection: "row",
    height: 20,
    justifyContent: "space-between",
  },
  carDescriptionContainer: {
    width: 233,
    paddingTop: 8,
    flexDirection: "row",
    justifyContent: "space-between",
    flexWrap: "wrap",
  },
  carDescriptionContainerText: {
    color: "#aaaaaa",
    fontSize: 13,
  },
  containerName: {
    paddingTop: 12,
    fontSize: 16,
    color: "#0288d1",
  },
  container: {
    marginRight: 14,
    width: 321,
    height: 170,
    backgroundColor: "#ffffff",
    borderRadius: 3,
    borderWidth: 2,
    borderColor: "#e1e1e1",
    justifyContent: "space-between",
  },
  containerOptions: {
    flexDirection: "row",
    height: 40,
    backgroundColor: "#f9f9f9",
    borderRadius: 2,
    justifyContent: "space-between",
  },
  containerOptionsElementLeft: {
    width: 120,
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "row",
  },
  containerOptionsElementRight: {
    width: 200,
    justifyContent: "center",
    flexDirection: "row",
    alignSelf: "center",
  },
  containerOptionsElementCenter: {
    alignSelf: "center",
    height: 20,
    width: 1,
    backgroundColor: "#e1e1e1",
  },
  containerOptionsText: {
    paddingLeft: 4,
    color: "#6f6f6f",
  },
});
