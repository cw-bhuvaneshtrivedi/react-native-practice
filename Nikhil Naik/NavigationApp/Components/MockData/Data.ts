export const info = [
  {
    city: "Mumbai",
    state: "Maharashtra",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque commodo in eros et bibendum. Vivamus facilisis quis tellus non tempus. Pellentesque hendrerit venenatis odio. Mauris egestas leo et nisi suscipit, ut blandit mauris pellentesque. Pellentesque a lacus eu felis efficitur tempor a eget dolor. Class aptent taciti sociosqu ad litora.",
    places: [
      "Marine Drive",
      "Gateway of India",
      "Shree Siddhivinayak",
      "Bandra-Worli Sea Link",
      "Chhatrapati Shivaji Terminus",
    ],
  },
  {
    city: "Kolkata",
    state: "West Bengal",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque commodo in eros et bibendum. Vivamus facilisis quis tellus non tempus. Pellentesque hendrerit venenatis odio. Mauris egestas leo et nisi suscipit, ut blandit mauris pellentesque. Pellentesque a lacus eu felis efficitur tempor a eget dolor. Class aptent taciti sociosqu ad litora.",
    places: [
      "Victoria Memorial Hall",
      "Dakshineswar Kali Temple",
      "Howrah Bridge",
      "Mother House",
      "Park Street",
    ],
  },
  {
    city: "Chennai",
    state: "Tamil Nadu",
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque commodo in eros et bibendum. Vivamus facilisis quis tellus non tempus. Pellentesque hendrerit venenatis odio. Mauris egestas leo et nisi suscipit, ut blandit mauris pellentesque. Pellentesque a lacus eu felis efficitur tempor a eget dolor. Class aptent taciti sociosqu ad litora.",
    places: [
      "Kapaleeshwarar Temple",
      "Snow Kingdom",
      "Besant Nagar Beach",
      "Marina Beach",
      "San Thome Church",
    ],
  },
];
