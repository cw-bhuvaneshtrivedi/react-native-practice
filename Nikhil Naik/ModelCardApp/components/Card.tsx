import {
  Text,
  View,
  StyleSheet,
  ScrollView,
  FlatList,
  ListRenderItemInfo,
  Dimensions,
} from "react-native";
import { InfoCard } from "./InfoCard";
import { DescriptionCard } from "./DescriptionCard";
import { verType } from "./Types/Type";
import { useState } from "react";
import { FlashList } from "@shopify/flash-list";

export const Card = ({ item }) => {
  const windowWidth = Dimensions.get("window").width;
  const [save, SetSave] = useState(0);
  return (
    <View>
      <View style={style.sep}></View>
      <View style={{ flexDirection: "row" }}>
        <View
          testID="sideBar"
          style={{
            flex: 1,
            height: 309,
            width: 4,
            backgroundColor: save > 0 ? "#00afa0" : "white",
          }}
        ></View>
        <View style={style.container}>
          <DescriptionCard
            img={item.hostUrl + "600x337" + item.originalImagePath}
            maker={item.makeName}
            model={item.modelName}
          />
          <FlashList
            data={item.versions}
            renderItem={(ver: ListRenderItemInfo<any>) => (
              <InfoCard ver={ver.item} border={SetSave} />
            )}
            horizontal={true}
            showsHorizontalScrollIndicator={false}
            estimatedItemSize={321}
            estimatedListSize={{
              height: 170,
              width: windowWidth,
            }}
            getItemType={(item) => item.name}
          />
          {/* <FlatList
            data={item.versions}
            renderItem={(ver: ListRenderItemInfo<any>) => (
              <InfoCard ver={ver.item} border={SetSave} />
            )}
            horizontal={true}
            showsHorizontalScrollIndicator={false}
          /> */}
        </View>
      </View>
    </View>
  );
};

const style = StyleSheet.create({
  container: {
    // backgroundColor: "#6f6f6f",
    backgroundColor: "white",
    height: 310,
    justifyContent: "flex-end",
    width: 405,
    paddingLeft: 17,
    paddingBottom: 21,
  },
  list: {
    flexDirection: "row",
    paddingRight: 17,
  },
  sep: {
    paddingLeft: 0,
    height: 4,
    backgroundColor: "#e1e1e1",
  },
});

{
  /* <InfoCard ver={ver} border={SetSave} data={save} /> */
}
