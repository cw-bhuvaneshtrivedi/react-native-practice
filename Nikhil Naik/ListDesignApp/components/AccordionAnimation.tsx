import { Dimensions, FlatList, StyleSheet, Text, View } from "react-native";
import React, { useEffect, useState } from "react";
import Card from "./Card";
import Animated, {
  useSharedValue,
  useAnimatedStyle,
  withTiming,
} from "react-native-reanimated";

export default function AccordionAnimation({
  open,
  data,
}: {
  open: boolean;
  data: string[];
}) {
  const [temp, setTemp] = useState(0);
  const height = useSharedValue(0);
  const style = useAnimatedStyle(() => ({
    height: height.value,
  }));
  useEffect(() => {
    if (open) {
      height.value = withTiming(temp);
    } else {
      height.value = withTiming(0);
    }
  }, [open]);
  return (
    <Animated.View style={[styles.container, style]} testID="card">
      <View style={{ position: "absolute" }}>
        <FlatList
          data={data}
          renderItem={({ item }) => <Card data={item} />}
          onLayout={(e) => setTemp(e.nativeEvent.layout.height)}
        />
      </View>
    </Animated.View>
  );
}

const styles = StyleSheet.create({
  container: { overflow: "hidden" },
});
