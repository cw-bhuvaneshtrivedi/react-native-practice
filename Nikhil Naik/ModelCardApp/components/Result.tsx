import { Card } from "./Card";
import { View, ScrollView, Text, FlatList, Dimensions } from "react-native";
import { FlashList } from "@shopify/flash-list";

export const Result = ({ data }) => {
  const windowWidth = Dimensions.get("window").width;
  const windowHeight = Dimensions.get("window").height;
  return (
    <View style={{ width: windowWidth, height: windowHeight }}>
      <FlashList
        data={data}
        renderItem={({ item }) => <Card item={item} />}
        estimatedItemSize={30}
      />
    </View>
  );
};
