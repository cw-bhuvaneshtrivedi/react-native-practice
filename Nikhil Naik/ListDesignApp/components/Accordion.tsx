import {
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import React, { useEffect, useState } from "react";
import { Feather } from "@expo/vector-icons";
import AccordionAnimation from "./AccordionAnimation";
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from "react-native-reanimated";

const Accordion = ({ data, idx, setView, view }) => {
  const [close, setClose] = useState(false);
  const rotateZ = useSharedValue(0);
  const style = useAnimatedStyle(() => ({
    transform: [{ rotateZ: `${rotateZ.value}` + "deg" }],
  }));
  const handle = () => {
    if (rotateZ.value == -180)
      rotateZ.value = withTiming(rotateZ.value + 180, { duration: 300 });
    else rotateZ.value = withTiming(rotateZ.value - 180, { duration: 300 });
  };
  useEffect(() => {
    if (close && view != -1 && view != idx) {
      handle();
      setClose(false);
    }
  }, [view]);
  const handlePress = () => {
    if (close) {
      setClose(!close);
      handle();
    } else {
      setClose(!close);
      handle();
      setView((item: number) => {
        return idx;
      });
    }
  };
  return (
    <ScrollView>
      <TouchableOpacity onPress={handlePress} testID="accordionButton">
        <View style={styles.container}>
          <Text style={styles.accordionText}>{data.makeName}</Text>
          <Animated.View style={[style]}>
            <Feather name="chevron-down" size={30} color="#aaaaaa" />
          </Animated.View>
        </View>
      </TouchableOpacity>
      <View style={styles.seperator}></View>
      <AccordionAnimation open={close} data={data} />
    </ScrollView>
  );
};

export default Accordion;

const styles = StyleSheet.create({
  container: {
    paddingLeft: 28,
    paddingRight: 30,
    height: 50,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  accordionText: {
    fontSize: 15,
    color: "#484848",
  },
  seperator: {
    width: "100%",
    height: 2,
    backgroundColor: "#efefef",
  },
});
