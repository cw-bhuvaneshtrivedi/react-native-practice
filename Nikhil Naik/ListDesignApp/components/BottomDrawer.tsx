import {
  Image,
  StyleSheet,
  Text,
  View,
  Dimensions,
  TextInput,
  TouchableOpacity,
} from "react-native";
import React, { useEffect } from "react";
import Animated, {
  useSharedValue,
  withTiming,
  useAnimatedStyle,
} from "react-native-reanimated";
import Accordion from "./Accordion";
import { AntDesign } from "@expo/vector-icons";

const HEIGHT = Dimensions.get("screen").height;
const WIDTH = Dimensions.get("screen").width;

const BottomDrawer = ({ open }: any) => {
  const translateY = useSharedValue(HEIGHT);
  const style = useAnimatedStyle(() => ({
    transform: [{ translateY: translateY.value }],
  }));
  const CloseEffect = () => {
    translateY.value = withTiming(HEIGHT, {
      duration: 300,
    });
    setTimeout(() => open(false), 250);
  };
  useEffect(() => {
    if (open) {
      translateY.value = withTiming(0, {
        duration: 300,
      });
    }
  }, [open]);
  return (
    <Animated.View style={[styles.container, style]}>
      <View style={styles.mainHeader}>
        <View style={[styles.header]}>
          <Text style={{ fontSize: 16, fontWeight: "bold", color: "#484848" }}>
            Select Your Make or Model
          </Text>
          <TouchableOpacity onPress={CloseEffect}>
            <AntDesign name="closecircleo" size={24} color="#484848" />
          </TouchableOpacity>
        </View>
        <TextInput
          placeholder="Type to Select Make "
          style={styles.searchBox}
        />
      </View>
      <Accordion />
      <Accordion />
      <Accordion />
      <Accordion />
    </Animated.View>
  );
};

export default BottomDrawer;

const styles = StyleSheet.create({
  container: {
    height: HEIGHT,
    backgroundColor: "white",
  },
  closeButtonText: {
    alignSelf: "flex-end",
    padding: 5,
    borderRadius: 50,
    borderStyle: "solid",
    borderColor: "grey",
    borderWidth: 2,
    alignItems: "center",
    justifyContent: "center",
  },
  searchBox: {
    borderRadius: 4,
    borderColor: "#d5d5d5",
    borderWidth: 2,
    borderStyle: "solid",
    width: "100%",
    marginTop: 10,
    height: 50,
    fontSize: 20,
    padding: 5,
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  mainHeader: {
    height: 120,
    paddingLeft: 16,
    paddingRight: 16,
    paddingTop: 16,
    marginBottom: 4,
  },
  bg: {
    // position: "absolute",
    width: WIDTH,
    height: 100,
  },
});
