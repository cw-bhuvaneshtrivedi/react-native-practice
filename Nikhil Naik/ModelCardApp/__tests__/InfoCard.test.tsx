import { fireEvent, render, screen } from "@testing-library/react-native";
import { InfoCard } from "../components/InfoCard";
import { data } from "../components/Mock/Data";

const mock = data[0].versions[0];

describe("Info Card", () => {
  it("should change text of compare option on button click", () => {
    render(<InfoCard ver={mock} border={jest.fn()} />);
    const result = screen.getByTestId("compareOption");
    const before = result.props.children[1].props.children;
    const button = screen.getByTestId("compareOptionButton");
    fireEvent(button, "click");
    const after = result.props.children[1].props.children;

    expect(before).toBe("Add to Compare");
    expect(after).toBe("Remove from Compare");
  });
});

describe("Info Card", () => {
  it("should change text of shortlist option on button click", () => {
    const setSave = jest.fn();
    setSave.mockImplementation((count) => count);
    render(<InfoCard ver={mock} border={setSave} />);

    const result = screen.getByTestId("shortlistOption");
    const before = result.props.children[1].props.children;
    const button = screen.getByTestId("shortlistOptionButton");
    fireEvent(button, "click");
    const after = result.props.children[1].props.children;

    expect(setSave).toHaveBeenCalled();
    expect(setSave).toHaveBeenCalledTimes(1);
    expect(before).toBe("Shortlist");
    expect(after).toBe("Shortlisted");
  });

  it("should match snapshot", () => {
    const result = render(<InfoCard ver={mock} border={jest.fn()} />);
    expect(result).toMatchSnapshot();
  });
});
