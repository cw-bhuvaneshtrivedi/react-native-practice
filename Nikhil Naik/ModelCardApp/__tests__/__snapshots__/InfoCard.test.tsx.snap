// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Card should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e1e1",
      "borderRadius": 3,
      "borderWidth": 2,
      "height": 170,
      "justifyContent": "space-between",
      "marginRight": 14,
      "width": 321,
    }
  }
>
  <View
    style={
      {
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#0288d1",
          "fontSize": 16,
          "paddingTop": 12,
        }
      }
      testID="carName"
    >
      VXi S-CNG
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "paddingTop": 8,
          "width": 233,
        }
      }
    >
      <Text
        style={
          {
            "color": "#aaaaaa",
            "fontSize": 13,
          }
        }
      >
        56 bhp
      </Text>
      <Text
        style={
          {
            "color": "#aaaaaa",
            "fontSize": 13,
          }
        }
      >
        998 cc
      </Text>
      <Text
        style={
          {
            "color": "#aaaaaa",
            "fontSize": 13,
          }
        }
      >
        Manual
      </Text>
      <Text
        style={
          {
            "color": "#aaaaaa",
            "fontSize": 13,
          }
        }
      >
         CNG 
      </Text>
    </View>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            {
              "color": "#484848",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Rs. 6,91,983
        </Text>
        <Text
          style={
            {
              "alignSelf": "flex-end",
              "color": "#0288d1",
              "fontSize": 13,
            }
          }
        >
          Show price in my city
        </Text>
        <Text
          style={
            {
              "alignSelf": "flex-end",
              "color": "#aaaaaa",
              "fontSize": 13,
              "fontStyle": "italic",
            }
          }
        >
          EMI
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "height": 20,
            "justifyContent": "space-between",
            "paddingTop": 4,
          }
        }
      >
        <Text
          style={
            {
              "alignSelf": "flex-end",
              "color": "#aaaaaa",
              "fontSize": 12,
            }
          }
        >
          Avg.Ex-Showroom Price
        </Text>
        <Text
          style={
            {
              "fontStyle": "italic",
            }
          }
        >
          Rs. 12,497
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f9f9f9",
        "borderRadius": 2,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="shortlistOptionButton"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": 120,
          }
        }
        testID="shortlistOption"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#6f6f6f",
                "fontSize": 15,
              },
              undefined,
              {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#6f6f6f",
              "paddingLeft": 4,
            }
          }
        >
          Shortlist
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#e1e1e1",
          "height": 20,
          "width": 1,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="compareOptionButton"
    >
      <View
        style={
          {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": 200,
          }
        }
        testID="compareOption"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#6f6f6f",
                "fontSize": 13,
              },
              undefined,
              {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#6f6f6f",
              "paddingLeft": 4,
            }
          }
        >
          Add to Compare
        </Text>
      </View>
    </View>
  </View>
</View>
`;
